go/mysqlconn/conn.go

```diff
@@ -256,7 +259,9 @@ func (c *Conn) writeOKPacket(affectedRows, lastInsertID uint64, flags uint16, wa
 	if err := c.writePacket(data); err != nil {
 		return err
 	}
-	c.flush()
+	if err := c.flush(); err != nil {
```

Comment:
Actually, for correctness, the unit tests catch it right away: no flush, the other side doesn't get the packet (happened a few times during development). So I don't think that's much of a concern.